wpiutil/generate_nanopb.py

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env python3
+
+# Copyright (c) FIRST and other WPILib contributors.
+# Open Source Software; you can modify and/or share it under the terms of
+# the WPILib BSD license file in the root directory of this project.
+import subprocess
+import sys
+import argparse
+import os
+import sys
+import shutil
+from pathlib import Path
+
+
+def generate_nanopb(nanopb: Path, output_directory: Path, proto_dir: Path):
+    shutil.rmtree(output_directory.absolute(), ignore_errors=True)
+    os.makedirs(output_directory.absolute())
+
+    proto_files = proto_dir.glob("*.proto")
+    for path in proto_files:
+        absolute_filename = path.absolute()
+        subprocess.run(
+            [
+                sys.executable,
+                nanopb,
+                f"-I{absolute_filename.parent}",
+                f"-D{output_directory.absolute()}",
+                "-S.cpp",
+                "-e.npb",
+                absolute_filename,
+            ]
+        )
+    java_files = (output_directory).glob("*")
+    for java_file in java_files:
+        with (java_file).open(encoding="utf-8") as f:
+            content = f.read()
+
+        java_file.write_text(
+            "// Copyright (c) FIRST and other WPILib contributors.\n// Open Source Software; you can modify and/or share it under the terms of\n// the WPILib BSD license file in the root directory of this project.\n"
+            + content,
+            encoding="utf-8",
+        )
+
+
+def main(argv):
```

Comment:
Should the `argv` argument should be removed per #7311?

Reply:
Yeah I need to go through those PRs and see what changed.

Reply:
Done